### PR TITLE
Make code readable

### PR DESCRIPTION
**Pull request by TylerVigario**
Working on understanding the codebase and making improvements.

**Changes:**

- ~~Simplify return object of `splitNum`~~
- ~~Standardize available functions & variables for language developers~~
- Document code for use with development tools (i.e. using JSDoc & IntelliSense)
- Improve local debugging for developers
- Performance improvements

### CODE REVIEW

File: lib/classes/N2WordsAbs.mjs [hunk 1]
```diff
@@ -54,25 +60,39 @@ export default class {
...
+
     let words = [];
-    let positiveValue = Math.abs(value);
+
+    // Convert value to positive value (if negative)
+    const positiveValue = Math.abs(value);
+
+    // Check if number Whole number or the language seperator word is undefined
+    // TODO: Figure out why seperator word matters
     if (value % 1 === 0 || typeof this.separatorWord === 'undefined') {
```
[the remark this answers is not in the document]
Comment (TylerVigario) [reply lib/classes/N2WordsAbs.mjs:71]: @forzagreen Thank you for explaining that! It struck me as odd to do that check so I left a note to further research. I'm hoping to have enough time today to finish the code review portion of my goal.

File: lib/classes/BaseLanguage.mjs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      }
+
+      // Is value perfect match of card number?
+      if (div == 1) {
+        // TODO: Merge word set pairs together (if possible) to simplify return object
+        out.push({
+          [this.getCardWord(1)]: 1,
+        });
+      } else {
+        // TODO: Understand the logic for this
```
Comment (TylerVigario) [inline lib/classes/BaseLanguage.mjs:58]: Could you possibly help me understand what's going on here?

Comment (TylerVigario) [inline lib/classes/BaseLanguage.mjs:58]: @forzagreen Could you possibly help me understand what's going on here?

Comment (forzagreen) [reply lib/classes/BaseLanguage.mjs:58]: @TylerVigario I'll try to remember and let you know

**Comment by TylerVigario** [comment]
@forzagreen After many attempts to recreate `splitNum()` logic (in a developer "friendly" way) I have realized two significant hurdles you overcame.

1. _Matching the highest value card creates a new number to match_ **(nested processing)** (i.e. 563 "thousand" for 563000)
2. Each language has requirements for using joining words/characters

It was my goal to break `splitNum` into smaller (defined) functions that are easy to understand. However, my current attempts break the current logic (which I don't currently fully understand).

**I will delay these logic changes until further discussion in a future PR.**

File: lib/classes/BaseLanguage.mjs (hunk 2 above)
Comment (forzagreen) [reply lib/classes/BaseLanguage.mjs]: I didn't find the details of the implementation. But I remember this "cards" part was inspired from [num2words](https://github.com/savoirfairelinux/num2words/blob/master/num2words/base.py)

**Comment by TylerVigario** [comment]
@forzagreen As long as this is okay with you, this should be good to merge.

**Review by forzagreen — APPROVED** [review]
@TylerVigario  LGTM
Thanks a lot ! 👏 👍

**Pull request merged**